netket/operator/_kinetic.py

```diff
@@ -0,0 +1,54 @@
+from typing import Optional, Union
+
+from netket.utils.types import DType, PyTree
+
+from netket.hilbert import AbstractHilbert
+from netket.operator import ContinousOperator
+
+import jax
+import jax.numpy as jnp
+
+
+class KineticEnergy(ContinousOperator):
+    r"""Returns the local kinetic energy (hbar = 1)
+                :math:`E_{kin} = -1/2 ( \sum_i \frac{1}{m_i} (\log(\psi))'^2 + (\log(\psi))'' )`
+    Args:
+    mass: float if all masses are the same, list indicating the mass of each particle otherwise
+    """
+
+    def __init__(
+        self,
+        hilbert: AbstractHilbert,
+        mass: Union[float, list],
+        dtype: Optional[DType] = float,
+    ):
+
+        self._mass = mass
+        self._dtype = dtype
+
+        super().__init__(hilbert)
+
+    @property
+    def dtype(self) -> DType:
+        return self._dtype
+
```

Comment:
maybe a better name...
```suggestion

    @property
    def mass(self):
        return self._mass
```